### ISSUE TITLE
Openshift Origin Master api service keeps on restart

### ISSUE THREAD

**Issue by riponbanik** [openshift/openshift-ansible#8130]
#### Description

Provide a brief description of your issue here. For example:

It is a single master and 2 nodes deployment using ref in using rpm package instead of docker image

https://docs.openshift.org/latest/install_config/install/advanced_install.html#single-master

##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`

2.4.0

If you're operating from a **git clone**:

* The output of `git describe`

openshift-ansible-3.9.19-1-26-gd328e7dda

If you're running from playbooks installed via RPM

* The output of `rpm -q openshift-ansible`

Place the output between the code block below:

```
VERSION INFORMATION HERE PLEASE
```

##### Steps To Reproduce
After running via Ansible master fails to start and connect via port 8443. By running the following command in controller it shows the api status as activating(start) instead of running.

```
sudo systemctl list-units | grep origin

  origin-master-api.service                                                                loaded activating start     start Atomic OpenShift Master API
  origin-master-controllers.service                                                        loaded active     running         Atomic OpenShift Master Controllers

sudo  journalctl -xe -fu origin-master-api.service

Apr 25 04:06:53 oc-master.lab.local systemd[1]: origin-master-api.service: main process exited, code=exited, status=255/n/a
Apr 25 04:06:53 oc-master.lab.local systemd[1]: Failed to start Atomic OpenShift Master API.
-- Subject: Unit origin-master-api.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit origin-master-api.service has failed.
--
-- The result is failed.
Apr 25 04:06:53 oc-master.lab.local systemd[1]: Unit origin-master-api.service entered failed state.
Apr 25 04:06:53 oc-master.lab.local systemd[1]: origin-master-api.service failed.
Apr 25 04:06:58 oc-master.lab.local systemd[1]: origin-master-api.service holdoff time over, scheduling restart.
Apr 25 04:06:58 oc-master.lab.local systemd[1]: Starting Atomic OpenShift Master API...
-- Subject: Unit origin-master-api.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit origin-master-api.service has begun starting up.
```

##### Expected Results
Describe what you expected to happen.

It needs to be like below as controller 

```
sudo systemctl status origin-master-controllers.service

● origin-master-controllers.service - Atomic OpenShift Master Controllers
   Loaded: loaded (/usr/lib/systemd/system/origin-master-controllers.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2018-04-25 02:23:55 UTC; 1h 13min ago
     Docs: https://github.com/openshift/origin
 Main PID: 3972 (openshift)
   CGroup: /system.slice/origin-master-controllers.service
           └─3972 /usr/bin/openshift start master controllers --config=/etc/origin/master/master-config.yaml --loglevel=2 --listen=https://0.0.0.0:8444

Apr 25 03:36:56 oc-master.lab.local origin-master-controllers[3972]: E0425 03:36:56.522851    3972 leaderelection.go:224] error retrieving resource lock kube-system/k... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.554801    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.554989    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.556685    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/kubernetes/plug...
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.558156    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.558256    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-go/infor...
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.559552    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.566552    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.611495    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Apr 25 03:37:03 oc-master.lab.local origin-master-controllers[3972]: E0425 03:37:03.614304    3972 reflector.go:205] github.com/openshift/origin/vendor/k8s.io/client-... timeout
Hint: Some lines were ellipsized, use -l to show in full.
```

##### Observed Results
Describe what is actually happening.

```
sudo systemctl status origin-master-api.service -l

● origin-master-api.service - Atomic OpenShift Master API
   Loaded: loaded (/usr/lib/systemd/system/origin-master-api.service; enabled; vendor preset: disabled)
   Active: activating (start) since Wed 2018-04-25 03:28:53 UTC; 12s ago
     Docs: https://github.com/openshift/origin
 Main PID: 5614 (openshift)
   CGroup: /system.slice/origin-master-api.service
           └─5614 /usr/bin/openshift start master api --config=/etc/origin/master/master-config.yaml --loglevel=2 --listen=https://0.0.0.0:8443 --master=https://oc-master.lab.local:8443

Apr 25 03:28:54 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:54.756780    5614 plugins.go:83] Registered admission plugin "SecurityContextDeny"
Apr 25 03:28:54 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:54.756874    5614 plugins.go:83] Registered admission plugin "ServiceAccount"
Apr 25 03:28:54 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:54.756916    5614 plugins.go:83] Registered admission plugin "DefaultStorageClass"
Apr 25 03:28:54 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:54.756978    5614 plugins.go:83] Registered admission plugin "PersistentVolumeClaimResize"
Apr 25 03:28:54 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:54.757043    5614 plugins.go:83] Registered admission plugin "PVCProtection"
Apr 25 03:28:56 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:56.536801    5614 master_config.go:445] Initializing cache sizes based on 0MB limit
Apr 25 03:28:56 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:56.537722    5614 master_config.go:509] Using the lease endpoint reconciler with TTL=15s and interval=10s
Apr 25 03:28:56 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:56.542423    5614 start_master.go:558] Starting master on 0.0.0.0:8443 (v3.9.0+ba7faec-1)
Apr 25 03:28:56 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:56.542543    5614 start_master.go:559] Public master address is https://oc-master.lab.local:8443
Apr 25 03:28:56 oc-master.lab.local atomic-openshift-master-api[5614]: I0425 03:28:56.542588    5614 start_master.go:566] Using images from "openshift/origin-<component>:v3.9.0"
```

For long output or logs, consider using a [gist](https://gist.github.com/)

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* Your operating system and version, ie: RHEL 7.2, Fedora 23 (`$ cat /etc/redhat-release`)
* Your inventory file (especially any non-standard configuration parameters)
* Sample code, etc

```
CentOS Linux release 7.4.1708 (Core)

**Inventory**

controller ansible_ssh_host=oc-master.lab.local 
node1      ansible_ssh_host=oc-node1.lab.local  
node2      ansible_ssh_host=oc-node2.lab.local  

[all:vars]
ansible_connection=ssh
ansible_ssh_user=vagrant
ansible_ssh_pass=vagrant

[nodes]
controller openshift_schedulable=false  
node[1:2] openshift_schedulable=true openshift_node_labels="{'region': 'primary', 'zone': 'default'}"


[OSEv3:children]
masters
nodes
etcd

[OSEv3:vars]
# SSH user, this user should allow ssh based auth without requiring a password
ansible_ssh_user=vagrant

# If ansible_ssh_user is not root, ansible_become must be set to true
ansible_become=true

# if your target hosts are Fedora uncomment this
#ansible_python_interpreter=/usr/bin/python3
openshift_deployment_type=origin

# uncomment the following to enable htpasswd authentication; defaults to DenyAllPasswordIdentityProvider
# openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': #'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]

# containerized=true
# openshift_release=3.9
# openshift_image_tag=latest 
# openshift_pkg_version=v3.9.0
# openshift_version=3.9


#osm_cluster_network_cidr=10.128.0.0/14
#openshift_portal_net=172.30.0.0/16
#osm_host_subnet_length=9

# localhost likely doesn't meet the minimum requirements
openshift_disable_check=disk_availability,docker_image_availability,memory_availability


# Access service externally
openshift_master_default_subdomain=192.168.56.30.xip.io

[masters]
controller

[etcd]
controller


```

**Comment by riponbanik** [comment]
I have found the issue related with default nat for vagrant, ctcd is running on eth0 by default and listening on 10.0.2.15. Is there any way to run master and etcd on all ip or specific interface?

Here are my config created by Ansible -

/etc/origin/master/master-config.yaml 

```
admissionConfig:
  pluginConfig:
    BuildDefaults:
      configuration:
        apiVersion: v1
        env: []
        kind: BuildDefaultsConfig
        resources:
          limits: {}
          requests: {}
    BuildOverrides:
      configuration:
        apiVersion: v1
        kind: BuildOverridesConfig
    openshift.io/ImagePolicy:
      configuration:
        apiVersion: v1
        executionRules:
        - matchImageAnnotations:
          - key: images.openshift.io/deny-execution
            value: 'true'
          name: execution-denied
          onResources:
          - resource: pods
          - resource: builds
          reject: true
          skipOnResolutionFailure: true
        kind: ImagePolicyConfig
apiLevels:
- v1
apiVersion: v1
controllerConfig:
  election:
    lockName: openshift-master-controllers
  serviceServingCert:
    signer:
      certFile: service-signer.crt
      keyFile: service-signer.key
controllers: '*'
corsAllowedOrigins:
  # anchor with start (\A) and end (\z) of the string, make the check case insensitive ((?i)) a
  - (?i)//127\.0\.0\.1(:|\z)
  - (?i)//localhost(:|\z)
  - (?i)//10\.0\.2\.15(:|\z)
  - (?i)//oc\-master\.lab\.local(:|\z)
  - (?i)//kubernetes\.default(:|\z)
  - (?i)//kubernetes\.default\.svc\.cluster\.local(:|\z)
  - (?i)//kubernetes(:|\z)
  - (?i)//openshift\.default(:|\z)
  - (?i)//openshift\.default\.svc(:|\z)
  - (?i)//172\.30\.0\.1(:|\z)
  - (?i)//openshift\.default\.svc\.cluster\.local(:|\z)
  - (?i)//kubernetes\.default\.svc(:|\z)
  - (?i)//openshift(:|\z)
dnsConfig:
  bindAddress: 0.0.0.0:8053
  bindNetwork: tcp4
etcdClientInfo:
  ca: master.etcd-ca.crt
  certFile: master.etcd-client.crt
  keyFile: master.etcd-client.key
  urls:
    - https://oc-master.lab.local:2379
etcdStorageConfig:
  kubernetesStoragePrefix: kubernetes.io
  kubernetesStorageVersion: v1
  openShiftStoragePrefix: openshift.io
  openShiftStorageVersion: v1
imageConfig:
  format: openshift/origin-${component}:${version}
  latest: False
kind: MasterConfig
kubeletClientInfo:
  ca: ca-bundle.crt
  certFile: master.kubelet-client.crt
  keyFile: master.kubelet-client.key
  port: 10250
kubernetesMasterConfig:
  apiServerArguments:
    storage-backend:
    - etcd3
    storage-media-type:
    - application/vnd.kubernetes.protobuf
  controllerArguments:
  masterCount: 1
  masterIP: 10.0.2.15
  podEvictionTimeout:
  proxyClientInfo:
    certFile: master.proxy-client.crt
    keyFile: master.proxy-client.key
  schedulerArguments:
  schedulerConfigFile: /etc/origin/master/scheduler.json
  servicesNodePortRange: ""
  servicesSubnet: 172.30.0.0/16
  staticNodeNames: []
masterClients:
  externalKubernetesClientConnectionOverrides:
    acceptContentTypes: application/vnd.kubernetes.protobuf,application/json
    contentType: application/vnd.kubernetes.protobuf
    burst: 400
    qps: 200
  externalKubernetesKubeConfig: ""
  openshiftLoopbackClientConnectionOverrides:
    acceptContentTypes: application/vnd.kubernetes.protobuf,application/json
    contentType: application/vnd.kubernetes.protobuf
    burst: 600
    qps: 300
  openshiftLoopbackKubeConfig: openshift-master.kubeconfig
masterPublicURL: https://oc-master.lab.local:8443
networkConfig:
  clusterNetworkCIDR: 10.128.0.0/14
  hostSubnetLength: 9
  clusterNetworks:
  - cidr: 10.128.0.0/14
    hostSubnetLength: 9
  networkPluginName: redhat/openshift-ovs-subnet
# serviceNetworkCIDR must match kubernetesMasterConfig.servicesSubnet
  serviceNetworkCIDR: 172.30.0.0/16
  externalIPNetworkCIDRs:
  - 0.0.0.0/0
oauthConfig:
  assetPublicURL: https://oc-master.lab.local:8443/console/
  grantConfig:
    method: auto
  identityProviders:
  - challenge: true
    login: true
    mappingMethod: claim
    name: allow_all
    provider:
      apiVersion: v1
      kind: AllowAllPasswordIdentityProvider
  masterCA: ca-bundle.crt
  masterPublicURL: https://oc-master.lab.local:8443
  masterURL: https://oc-master.lab.local:8443
  sessionConfig:
    sessionMaxAgeSeconds: 3600
    sessionName: ssn
    sessionSecretsFile: /etc/origin/master/session-secrets.yaml
  tokenConfig:
    accessTokenMaxAgeSeconds: 86400
    authorizeTokenMaxAgeSeconds: 500
pauseControllers: false
policyConfig:
  bootstrapPolicyFile: /etc/origin/master/policy.json
  openshiftInfrastructureNamespace: openshift-infra
  openshiftSharedResourcesNamespace: openshift
projectConfig:
  defaultNodeSelector: "node-role.kubernetes.io/compute=true"
  projectRequestMessage: ""
  projectRequestTemplate: ""
  securityAllocator:
    mcsAllocatorRange: "s0:/2"
    mcsLabelsPerProject: 5
    uidAllocatorRange: "1000000000-1999999999/10000"
routingConfig:
  subdomain:  "192.168.56.30.xip.io"
serviceAccountConfig:
  limitSecretReferences: False
  managedNames:
  - default
  - builder
  - deployer
  masterCA: ca-bundle.crt
  privateKeyFile: serviceaccounts.private.key
  publicKeyFiles:
  - serviceaccounts.public.key
servingInfo:
  bindAddress: 0.0.0.0:8443
  bindNetwork: tcp4
  certFile: master.server.crt
  clientCA: ca.crt
  keyFile: master.server.key
  maxRequestsInFlight: 500
  requestTimeoutSeconds: 3600
volumeConfig:
  dynamicProvisioningEnabled: True

```
sudo cat /etc/etcd/etcd.conf

```
ETCD_NAME=oc-master.lab.local
ETCD_LISTEN_PEER_URLS=https://10.0.2.15:2380
ETCD_DATA_DIR=/var/lib/etcd/
#ETCD_WAL_DIR=""
#ETCD_SNAPSHOT_COUNT=10000
ETCD_HEARTBEAT_INTERVAL=500
ETCD_ELECTION_TIMEOUT=2500
ETCD_LISTEN_CLIENT_URLS=https://10.0.2.15:2379
#ETCD_MAX_SNAPSHOTS=5
#ETCD_MAX_WALS=5
#ETCD_CORS=


#[cluster]
ETCD_INITIAL_ADVERTISE_PEER_URLS=https://10.0.2.15:2380
ETCD_INITIAL_CLUSTER=oc-master.lab.local=https://10.0.2.15:2380
ETCD_INITIAL_CLUSTER_STATE=new
ETCD_INITIAL_CLUSTER_TOKEN=etcd-cluster-1
#ETCD_DISCOVERY=
#ETCD_DISCOVERY_SRV=
#ETCD_DISCOVERY_FALLBACK=proxy
#ETCD_DISCOVERY_PROXY=
ETCD_ADVERTISE_CLIENT_URLS=https://10.0.2.15:2379
#ETCD_STRICT_RECONFIG_CHECK="false"
#ETCD_AUTO_COMPACTION_RETENTION="0"
#ETCD_ENABLE_V2="true"
ETCD_QUOTA_BACKEND_BYTES=4294967296

#[proxy]
#ETCD_PROXY=off
#ETCD_PROXY_FAILURE_WAIT="5000"
#ETCD_PROXY_REFRESH_INTERVAL="30000"
#ETCD_PROXY_DIAL_TIMEOUT="1000"
#ETCD_PROXY_WRITE_TIMEOUT="5000"
#ETCD_PROXY_READ_TIMEOUT="0"

#[security]
ETCD_TRUSTED_CA_FILE=/etc/etcd/ca.crt
ETCD_CLIENT_CERT_AUTH="true"
ETCD_CERT_FILE=/etc/etcd/server.crt
ETCD_KEY_FILE=/etc/etcd/server.key
#ETCD_AUTO_TLS="false"
ETCD_PEER_TRUSTED_CA_FILE=/etc/etcd/ca.crt
ETCD_PEER_CLIENT_CERT_AUTH="true"
ETCD_PEER_CERT_FILE=/etc/etcd/peer.crt
ETCD_PEER_KEY_FILE=/etc/etcd/peer.key
#ETCD_PEER_AUTO_TLS="false"

#[logging]
ETCD_DEBUG="False"

#[profiling]
#ETCD_ENABLE_PPROF="false"
#ETCD_METRICS="basic"
#
#[auth]
#ETCD_AUTH_TOKEN="simple"
```

When I change the above IP 10.0.2.15 with my private ip (192.168.56.30), the restart of the services works (etcd and master api).

But browsing console - https://oc-master.lab.local:8443/console/, shows the following errors -

missing service (service "webconsole" not found)
missing route (service "webconsole" not found)

**Comment by riponbanik** [comment]
Do you guys care about building a working software or is it just a platform for us to do trial and error so that you guys can fix the issue for the Enterprise? Fixing one issue and coming up with another. The same ansible run behaves differently at different time. It sucks!!!

**Comment by vrutkovs** [comment]
>Is there any way to run master and etcd on all ip or specific interface?

`openshift_dns_ip` might work

>But browsing console - https://oc-master.lab.local:8443/console/, shows the following errors -

Did the deploy finish? Were pods in `openshift-web-console` namespace created?

**Comment by riponbanik** [comment]
@vrutkovs Thanks. I know about that, although origin documents discourage using it as below. I have resolved the issue by seeing dnsIP instead in /etc/origin/node/node-config.yaml. Deployment did not finish but could now build image, upload to internal registry and deploy the container by manually modifying some config.

"Instead, allow the installer to configure each node to use dnsmasq and forward requests to the external DNS provider or SkyDNS, the internal DNS service for cluster-wide DNS resolution of internal hostnames for services and pods. If you do set the openshift_dns_ip option, then it should be set either with a DNS IP that queries SkyDNS first, or to the SkyDNS service or endpoint IP (the Kubernetes service IP)." 

Now see "openshift_service_catalog : wait for api server to be ready" failure, although previously it was fixed the issue by specifying -

controller openshift_schedulable=true openshift_node_labels="{'region': 'infra','zone': 'default'}"  

The main problem i see, 99-origin-dns.sh is not working although set the interface correctly as below and not proper documentation about a workable solution/build. I am doing trail and error for the last couple of months and been frustrated about it. 

cat /etc/sysconfig/network-scripts/ifcfg-ens33

............
NM_CONTROLLED=yes
PEERDNS=yes

........

**Comment by vrutkovs** [comment]
>I know about that, although origin documents discourage using it as below

It discouraged in general, but it seems to do what's required in this case

>I have fixed the issue by specifying 127.0.0.1 as DNS server of my interface since I see https://apiserver.kube-service-catalog.svc is what it is querying about.

127.0.0.1 is specified as a server in dnsmasq, in /etc/dnsmasq.d/ configs, is it being set there?

>Do not know how many problems more to come

Lets create one ticket per problem, for instance previous comments list at least three different not linked issues

**Comment by riponbanik** [comment]
ya sure.
